src/redux/shoppingCart.js

```diff
@@ -0,0 +1,76 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+// Helper function to save cart state to localStorage
+const saveCartToLocalStorage = (cartState) => {
+  localStorage.setItem('cartState', JSON.stringify(cartState));
+};
+
+// Helper function to load cart state from localStorage
+const loadCartFromLocalStorage = () => {
+  const savedState = localStorage.getItem('cartState');
+  return savedState ? JSON.parse(savedState) : undefined;
+};
+
+const initialState = loadCartFromLocalStorage() || {
+  items: {},
+  subtotal: 0,
+  tax: 0.15,
+  totalQuantity: 0,
+  totalAmount: 0,
+};
+
+const shoppingCartReducer = createSlice({
+  name: 'shoppingCart',
+  initialState,
+  reducers: {
+    addItem(state, action) {
+      const { album, quantity } = action.payload;
+      const albumId = album.id;
+
+      if (!state.items[albumId]) {
+        state.items[albumId] = { album, quantity: 0 };
+      }
+
+      state.items[albumId].quantity += quantity;
+      state.totalQuantity += quantity;
+      state.subtotal += album.price * quantity;
+      state.totalAmount = state.subtotal + state.subtotal * state.tax;
+
+      saveCartToLocalStorage(state);
+    },
+    reduceItem(state, action) {
+      const { album, quantity } = action.payload;
+      const albumId = album.id;
+
+      if (state.items[albumId]) {
+        const remainingQuantity = Math.max(
+          state.items[albumId].quantity - quantity,
+          0
+        );
+        state.totalQuantity -=
+          state.items[albumId].quantity - remainingQuantity;
```

Comment:
Hmm, if I'm reading this correctly I don't think it works in the following example case:

```javascript
{  
  items: [ {album: {id: "id1", ...etc}, quantity: 3 }],
  totalQuantity: 3
}
```

And a `reduceItem` event comes in with the payload:

```javascript
{ album: {id: "id1", ...etc}, quantity: 10 }
```

So `reamainingQuantity` will be 0. And then here on lines 50-51, we'd be doing 

`state.totalQuantity =`   
`state.totalQuantity - state.items["id1"].quantity - remainingQuantity =`   
`3 - (3 - 10) =`   
`-10`?